### PR TITLE
MC-127 - Office 365 Analysis Workflow Improvements

### PR DESCRIPTION
**Pull request by dmandyna-r7**
## Requirements

Workflow contributors, verify you have completed the following items by checking them off:

### Checklist
- [ ] Workflow verified as working in a test
- [ ] Workflow bundle includes `Name_of_Workflow.icon` file, `help.md` file, `workflow.spec.yaml` file, and `screenshots` directory
- [ ] Workflow title should follow general [Title of Articles](http://grammar.yourdictionary.com/capitalization/rules-for-capitalization-in-titles.html) format
- [ ] Workflow includes a detailed Description
- [ ] Workflow step names follow [Title of Articles](http://grammar.yourdictionary.com/capitalization/rules-for-capitalization-in-titles.html) format
- [ ] Triggers, Decisions, and other workflow steps include descriptions where needed
- [ ] Workflow includes at least one `Final Report` artifact that summarizes workflow execution details
- [ ] Workflow bundle (directory) is named after the title of the workflow in snake case with first letters capitalized, e.g. `workflows/Name_of_Workflow/`
- [ ] `help.md` file includes Description, Key Features, Requirements, Setup, Technical Details, Version History, and References
- [ ] `workflow.spec.yaml` file includes `products`, `name`, `title`, `description`, `use_cases` tags, `keywords` tags, `source_url`, and the `name` for each screenshot matches a PNG image in the `screenshots` directory

### Assessment

<details><summary>icon-validate ./Force_Password_Reset_in_Azure_AD_from_Microsoft_Teams</summary>

```
[*] Validating workflow at .\Office_365_Analysis\

[*] Running Integration Validators...
[*] Executing validator WorkflowDirectoryNameMatchValidator
[*] Executing validator WorkflowFilesValidator
[*] Executing validator WorkflowHelpValidator
[*] Executing validator WorkflowChangelogValidator
[*] Executing validator WorkflowVendorValidator
[*] Executing validator WorkflowVersionValidator
[*] Executing validator WorkflowExtensionValidator
[*] Executing validator WorkflowSupportValidator
[*] Executing validator WorkflowPNGHashValidator
[*] Executing validator WorkflowICONFileNameValidator
[*] Executing validator WorkflowScreenshotValidator
[*] Executing validator WorkflowTitleValidator
[*] Executing validator WorkflowDescriptionValidator
[*] Executing validator WorkflowNameValidator
[*] Executing validator WorkflowProfanityValidator
[*] Executing validator WorkflowHelpPluginUtilizationValidator
[*] Executing validator WorkflowICONFileValidator
[*] Workflow failed validation! The following validation errors occurred:

Validator "WorkflowDirectoryNameMatchValidator" failed!
        Cause: Workflow directory name does not match the workflow filename!


----
[*] Total time elapsed: 111.189ms
```
</details>

<details><summary>Screenshots</summary>

#### Workflow Builder
workflow1

workflow2

workflow3

workflow4

workflow5

workflow6

#### Artifact
artifact1

artifact2

#### Messages
notification1

</details>

### CODE REVIEW

**Review by ncramer-r7 — COMMENTED** [review]
VERY minor thing, but there is a reference to Jon Schipp in an artifact screenshot. Everything looks good though, so I'll approve after that is sanitized!

**Comment by dmandyna-r7** [comment]
Thanks for picking this up, I've updated the screenshot now.

**Review by ncramer-r7 — APPROVED** [review]

File: workflows/Office_365_Analysis/Office_365_Analysis.icon [hunk 1]
```diff
@@ -1074,1424 +1123,2094 @@
...
+          },
+          "1b82897d-95a2-4799-87d8-881d201d8fa8": {
+            "nodeId": "1b82897d-95a2-4799-87d8-881d201d8fa8",
+            "name": "# Remediation",
+            "type": "artifact",
+            "continueOnFailure": false,
+            "isDisabled": false,
+            "parameters": {
+              "input": {
+                "content": "# Remediation Goes Here\n\n(this artifact can be removed)\n\nAt this point in the workflow, we've established we have a malicious email. \n\nRemediation steps go on this branch of the flow. They can be email notifications, slack notifications for human follow up. Remediation can also go further and block the sender and delete the email from your organization. \n\nIn this example, I will send a notification via email, and use the Office Admin plugin to both delete the email and block the sender from our organization. "
```
Comment (mrinehart-r7) [inline workflows/Office_365_Analysis/Office_365_Analysis.icon:1187]:
```suggestion
                "content": "# Remediation Goes Here\n\n(this artifact can be removed)\n\nAt this point in the workflow, we've established we have a malicious email. \n\nRemediation steps go on this branch of the flow. They can be email notifications, Slack notifications for human follow up. Remediation can also go further and block the sender and delete the email from your organization. \n\nIn this example, I will send a notification via email, and use the Office Admin plugin to both delete the email and block the sender from our organization. "
```

File: workflows/Office_365_Analysis/Office_365_Analysis.icon [hunk 2]
```diff
@@ -1074,1424 +1123,2094 @@
...
-            "name": "Is Email Malicious",
+          "c34206e6-bd3c-4b35-b45d-ceb6281f373d": {
+            "nodeId": "c34206e6-bd3c-4b35-b45d-ceb6281f373d",
+            "name": "Attached Email Analysis Overview",
+            "type": "artifact",
+            "continueOnFailure": false,
+            "isDisabled": false,
+            "parameters": {
+              "input": {
+                "content": "## Attached Email Analysis\n\n(This artifact can be removed) \n\nThis loop will analyze any each of the attached emails and look for any malicious indicators - these include any malicious URLs found within these emails, any attached files to these emails and if the email comes from a domains that is younger than 30 days. This will be the best indicators of a suspicious email. If any are found this email will be marked as malicious. "
```
Comment (mrinehart-r7) [inline workflows/Office_365_Analysis/Office_365_Analysis.icon:2951]:
```suggestion
                "content": "## Attached Email Analysis\n\n(This artifact can be removed) \n\nThis loop will analyze any each of the attached emails and look for any malicious indicators - these include any malicious URLs found within these emails, any attached files to these emails and if the email comes from a domains that is younger than 30 days. These will be the best indicators of a suspicious email. If any are found this email will be marked as malicious. "
```

File: workflows/Office_365_Analysis/help.md [hunk 3]
```diff
@@ -10,16 +10,21 @@ This workflow is designed to automatically analyze incoming emails to determine
...
+* Hybrid Analysis
 
 # Documentation
 
 ## Setup
 
-Once the workflow has been downloaded, login to InsightConnect and “Import” it into the workflow builder. Once imported, you will initially be prompted to configure the connections for each of the plugins.
+Import the workflow from the Rapid7 Extension Library and proceed through the Import Workflow wizard in InsightConnect. Import plugins, create or select connections, and rename the workflow as a part of the Import Workflow wizard as necessary.
+
+Once the workflow has been imported. **update the first step with the mailbox and subject to suit your environment** by editing the input with the preset text of `change_me`.
```
Comment (mrinehart-r7) [inline workflows/Office_365_Analysis/help.md:25]:
```suggestion
Once the workflow has been imported, **update the first step with the mailbox and subject to suit your environment** by editing the input with the preset text of `change_me`.
```

**Review by mrinehart-r7 — CHANGES_REQUESTED** [review]
Few minor changes. Workflow validation is also failing for some reason.

**Comment by dmandyna-r7** [comment]
> Few minor changes. Workflow validation is also failing for some reason.

Hey @mrinehart-r7,

The workflow validation is failing because the name of the folder where we store the workflow is not the same as workflow filename. Let me know if that's fine as it is (I didn't change any file/folder names as part of this ticket), or if you would like me to change either the folder name or the workflow file name (if so, let me know which one to change) 🙂 

Thank you!

**Review by jschipp-r7 — APPROVED** [review]

**Comment by jschipp-r7** [comment]
> > Few minor changes. Workflow validation is also failing for some reason.
> 
> Hey @mrinehart-r7,
> 
> The workflow validation is failing because the name of the folder where we store the workflow is not the same as workflow filename. Let me know if that's fine as it is (I didn't change any file/folder names as part of this ticket), or if you would like me to change either the folder name or the workflow file name (if so, let me know which one to change) 🙂
> 
> Thank you!

Looks like the name is the same (requirement) but the title is different so we should be good here.

**Pull request merged**